### PR TITLE
Implement vtex deps diff command

### PR DESCRIPTION
**Pull request by vcalasans**
This command shows a diff of dependencies between the first and second arguments.

Usage is `vtex deps diff <workspace1>(optional) <workspace2>(optional)`.

Default value for `workspace1` is `'master'`. Default value for `workspace2` is `{current workspace}`.

image

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by klzns** [comment]
Why is there a `-` sign beside all apps on the left?

**Comment by vcalasans** [comment]
> Why is there a `-` sign beside all apps on the left?

These are removed dependencies. The `-` is to show that, in order to 'get' to the workspace on the left, you need to remove the dependencies prefixed with `-` and add the ones prefixed with `+`. Do you think this is unnecessary?

File: src/modules/tree.ts [hunk 1]
```diff
@@ -57,6 +57,12 @@ export default {
       handler: './deps/update',
       optionalArgs: ['app'],
     },
+    diff: {
+      alias: 'diff',
```
[the remark this answers is not in the document]
Comment (vcalasans) [reply src/modules/tree.ts]: As agreed, removed the alias.

File: src/modules/deps/utils.ts [hunk 2]
```diff
@@ -11,3 +17,130 @@ export const removeNpm = (deps, inValues?) => {
   })
   return deps
 }
+
+export const cleanDeps = R.compose(R.keys, removeNpm)
+
+export const getCleanDependencies = async (context) => {
+return await new Apps(context)
```
[the remark this answers is not in the document]
Comment (vcalasans) [reply src/modules/deps/utils.ts]: Done

**Comment by vcalasans** [comment]
> As discussed in person: one workspace can have different versions of the same app as dependecies. This implementation ignore this case.

Solved.

**Pull request merged**